src/components/asset-amount-input/asset-amount-input.tsx

```diff
@@ -163,7 +163,20 @@ export const AssetAmountInput: FC<AssetAmountInputProps> = ({
             equalityFn={tokenEqualityFn}
             renderValue={renderTokenValue}
             renderListItem={renderTokenListItem}
-            onValueChange={newAsset => onValueChange({ ...value, asset: newAsset ?? emptyToken })}
+            onValueChange={newAsset => {
+              onValueChange({
+                amount: isDefined(inputValueRef.current)
+                  ? isTokenInputType
+                    ? tzToMutez(inputValueRef.current, newAsset?.decimals ?? 0)
+                    : dollarToTokenAmount(
```

Comment:
very (cognitively) hard condition

Reply:
consider to refactor to some `formatAmount(ref, normalize(decimals), rates)` kind of function